tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -6837,34 +6890,41 @@ uint32_t SPIRVEmitter::tryToEvaluateAsFloat32(const llvm::APFloat &floatValue) {
 
 uint32_t SPIRVEmitter::translateAPFloat(const llvm::APFloat &floatValue,
                                         QualType targetType) {
+  // The float value may have to go through conversion, so work on a local copy.
+  llvm::APFloat value = floatValue;
```

Comment:
Nit: Actually we can change the signature to take in a `llvm::APFloat` instead of `const llvm::APFloat&`. That will handle the copy implicitly.